blocks/library/paragraph/index.js

```diff
@@ -160,10 +160,7 @@ class ParagraphBlock extends Component {
 				</InspectorControls>
 			),
 			<div key="editable" ref={ this.bindRef }>
-				<Autocomplete completers={ [
-					blockAutocompleter( { onReplace } ),
-					userAutocompleter(),
-				] }>
+				<ContextualAutocomplete onReplace={ onReplace }>
 					{ ( { isExpanded, listBoxId, activeId } ) => (
 						<RichText
```

Comment:
Can we wrap `RichText` with `ContextualAutocomplete`? We have now access to autocompleters through context so this should work :)

Reply:
Definitely something to think about in the future. Let’s see how many RichText components will we wrap. It’d be more useful for plugin developers as they would easily update core components, but sometimes there might be a few RichText components bundled together, like in case of `Quote`.

Reply:
> Can we wrap RichText with ContextualAutocomplete? We have now access to autocompleters through context so this should work :)

I was thinking about that too because I wanted block authors to get autocompletion without manually wrapping their inputs with an autocomplete component. I wonder if we should just define `ContextualAutocomplete` within `RichText` and use it to directly wrap `<TinyMCE>`.

That brings to mind another lingering question:
Should we support declaring default autocompleters?

I'm thinking the user autocompleter would be desired everywhere RichText is used, while the block autocompleter could be odd or surprising.

Reply:
Yes, it raises more questions. I think we should continue as it is, but revisit later when we have more blocks using this feature.

Reply:
I didn't have time to spend on this today, but for a good first implementation of extensible autocompletion, I'm thinking we should do the following and then polish:

* Move `<ContextualAutocomplete>` out of the paragraph block and into `RichEdit`, wrapping `TinyMCE`. I believe this would give us fairly good autocompletion support out of the box.
* Provide a set of default autocompleters that are assigned to each registered block without declared autocompleters. It would probably be just the user completer for now.
  * We could support filtering the default set of autocompleters later.
  * I don't love actually applying the defaults to the block settings, but while we aren't providing a default autocompleters filter, it would allow a `blocks.registerBlockType` filter to override autocompleters for any block type.

You mentioned
> I think we should continue as it is
but I'm not 100% whether that means you'd rather not do the items above. What do you think?